cal/utils.py

```diff
@@ -82,3 +87,26 @@ def get_headers(resp):
 
     if request_id not in resource.req_ids:
         resource.req_ids.append(request_id)
+
+
+def pick_cloud_provider():
+    # Random pick one cloud provider.
+    provider = list(CONF.providers.driver_mapper.keys())
+    return random.choice(provider)
+
+
+def pick_host_with_specific_provider(provider, cloud_config=None):
+    if cloud_config is None:
+        hosts = getattr(CONF, provider.lower())['hosts']
+        # Now, random choice host from provider hosts.
+        # TODO(kiennt): Next phase, pick the most optimized host
+        #               of given provider.
+        picked_host_config = hosts[random.choice(list(hosts.keys()))]
```

Comment:
I have a bad feeling with random lib, but OK because you realize it via TODO note.
Next, please implement a filter mechanism.